### ISSUE TITLE
MemberNotNull should supress warning CS8618 when called in constructor

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#49493]
**Version Used**: 

**Steps to Reproduce**:

```csharp
using System.Diagnostics.CodeAnalysis;
#nullable enable
public class C {
    public void M() {
       Init();
       S.ToString();
    }
    
    [MemberNotNull(nameof(S))]
    public void Init(){ S = ""; }
    
    public string S { get; set; }
}

namespace System.Diagnostics.CodeAnalysis
{
    [AttributeUsage(AttributeTargets.Method | AttributeTargets.Property, Inherited = false, AllowMultiple = true)]
    public sealed class MemberNotNullAttribute : Attribute
    {
        /// <summary>Initializes the attribute with a field or property member.</summary>
        /// <param name="member">
        /// The field or property member that is promised to be not-null.
        /// </param>
        public MemberNotNullAttribute(string member) => Members = new[] { member };

        /// <summary>Initializes the attribute with the list of field and property members.</summary>
        /// <param name="members">
        /// The list of field and property members that are promised to be not-null.
        /// </param>
        public MemberNotNullAttribute(params string[] members) => Members = members;

        /// <summary>Gets field or property member names.</summary>
        public string[] Members { get; }
    }
}
```

**Expected Behavior**: No warning

**Actual Behavior**: 
```
warning CS8618: Non-nullable property 'S' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
```

**Comment by RikkiGibson** [comment]
It appears that 'Init' was not used in a constructor in the given sample. Changing 'void M' to 'C' gives the expected behavior.

**Comment by YairHalberstadt** [comment]
Oops! 😱
